src/components/ChatApp/MessageSection/MessageSection.react.js

```diff
@@ -219,111 +147,21 @@ class MessageSection extends Component {
-    let messages = this.state.messages;
-    let markingData = searchMsgs(
-      messages,
-      matchString,
-      this.state.searchState.caseSensitive,
-    );
-
-    if (matchString) {
-      let searchState = {
-        markedMsgs: markingData.allmsgs,
-        markedIDs: markingData.markedIDs,
-        markedIndices: markingData.markedIndices,
-        scrollLimit: markingData.markedIDs.length,
-        scrollIndex: 0,
-        scrollID: markingData.markedIDs[0],
-        caseSensitive: this.state.searchState.caseSensitive,
-        open: false,
-        searchIndex: 1,
-        searchText: matchString,
-      };
-      if (markingData.markedIDs.length === 0 && matchString.trim().length > 0) {
-        searchState.searchIndex = 0;
-        this.props.openSnackBar({
-          snackBarMessage: 'No Results!',
-        });
-      }
-      this.setState({
-        searchState,
-      });
-    } else {
-      let searchState = {
-        markedMsgs: markingData.allmsgs,
-        markedIDs: markingData.markedIDs,
-        markedIndices: markingData.markedIndices,
-        scrollLimit: 0,
-        scrollIndex: -1,
-        scrollID: null,
-        caseSensitive: this.state.searchState.caseSensitive,
-        open: false,
-        searchIndex: 0,
-        searchText: '',
-      };
-      this.setState({
-        searchState: searchState,
-      });
-    }
-  };
-
-  componentDidMount() {
-    switch (this.state.currTheme) {
-      case 'light': {
-        document.body.className = 'white-body';
-        break;
-      }
-      case 'dark': {
-        document.body.className = 'dark-body';
-        break;
-      }
-      default: {
-        // do nothing
-      }
-    }
-
-    UserPreferencesStore.on('change', () => {
-      this.setState({
-        currTheme: UserPreferencesStore.getTheme(),
-      });
-      switch (this.state.currTheme) {
-        case 'light': {
-          document.body.className = 'white-body';
-          break;
-        }
-        case 'dark': {
-          document.body.className = 'dark-body';
-          break;
-        }
-        default: {
-          // do nothing
-        }
-      }
-    });
-
-    this._scrollToBottom();
-    MessageStore.addChangeListener(this._onChange);
-    ThreadStore.addChangeListener(this._onChange);
-  }
-
-  // Scroll to bottom feature goes here
   onScroll = () => {
-    let scrollarea = this.scrollarea;
-    if (scrollarea) {
-      let scrollValues = scrollarea.getValues();
+    const { showScrollBottom, showScrollTop } = this.state;
+    if (this.scrollarea) {
+      let scrollValues = this.scrollarea.getValues();
       if (scrollValues.top === 1) {
         this.setState({
+          showScrollTop: true,
           showScrollBottom: false,
         });
       } else if (scrollValues.top === 0) {
         this.setState({
           showScrollTop: false,
           showScrollBottom: true,
         });
-      } else {
+      } else if (!(showScrollBottom && showScrollTop)) {
```

Comment:
This optimization will save use tons of re-renders. And now the scroll works very smooth.